docs/Validation-and-Serialization.md

```diff
@@ -658,7 +658,92 @@ fastify.setErrorHandler(function (error, request, reply) {
 })
 ```
 
-If you want custom error response in schema without headaches and quickly, you can take a look at [here](https://github.com/epoberezkin/ajv-errors)
+If you want custom error response in schema without headaches and quickly, you can take a look at [`ajv-errors`](https://github.com/epoberezkin/ajv-errors).
+
+Below is an example showing how to add custom error messages for each property of a schema by supplying a custom AJV instance. Inline comments describe how to configure the schema to show a different error message for each case:
+
+```js
+const Ajv = require('ajv')
+const ajvErrors = require('ajv-errors')
+
+const schema = {
+  body: {
+    type: 'object',
+    properties: {
+      name: {
+        type: 'string',
+        errorMessage: {
+          type: 'Bad name'
+        }
+      },
+      age: {
+        type: 'number',
+        errorMessage: {
+          type: 'Bad age', // specify custom message for
+          min: 'Too young' // all constraints except required
+        }
+      }
+    },
+    required: ['name', 'age'],
+    errorMessage: {
+      required: {
+        name: 'Why no name!', // specify error message for when the
+        age: 'Why no age!' // property is missing from input
+      }
+    }
+  }
+}
+
+const ajv = new Ajv({ allErrors: true, jsonPointers: true })
+// extend the AJV instance with custom errors support
+ajvErrors(ajv)
+
+fastify.setSchemaCompiler(function (schema) {
+  return ajv.compile(schema)
+})
+```
+
+If you want to return localized error messages, take a look at [ajv-i18n](https://github.com/epoberezkin/ajv-i18n)
+
+```js
+const Ajv = require('ajv')
+const localize = require('ajv-i18n')
+
+const schema = {
+  body: {
+    type: 'object',
+    properties: {
+      name: {
+        type: 'string',
+      },
+      age: {
+        type: 'number',
+      }
+    },
+    required: ['name', 'age'],
+  }
+}
+
+const ajv = new Ajv({ allErrors: true, jsonPointers: true })
+
+fastify.setSchemaCompiler(function (schema) {
+  const validate =  ajv.compile(schema)
+
+  function validator (data) {
+    const result = validate(data)
+
+    if (!result) {
+      localize.ru(validate.errors);
+      ajv.errorsText(validate.errors, { separator: '\n' })
```

Comment:
I would manage this in a different manner after creating the `fastify` instance as above.
Less code to maintain for the user and it could become a plugin to reuse in other application

```
fastify.setErrorHandler(function (error, request, reply) {
  if (error.validation) {
     localize.ru(error.validation)
     fastify.schemaCompiler.errorsText(error.validation, { separator: '\n' })
     reply.status(400).send(error.validation)
  }
})
```

Reply:
`errorsText` is not on `fastify.schemaCompiler`. It's on `ajv`'s instance. Whereas `fastify.schemaCompiler` is a compiler's instance, which is a function not an object.

So I've kept the custom ajv instance and using it in `errorHandler`. Let me know if you have any other way of accessing ajv's instance.

Reply:
You are right, but we can just remove the `fastify.schemaCompiler.errorsText(e` line
We were building a string without use it.

Reply:
Damn! That's silly 🤦 I assumed that line is also mutating the errors object just like the one above. Removed 😅